netdata-installer.sh

```diff
@@ -413,6 +421,74 @@ fi
 
 trap build_error EXIT
 
+# -----------------------------------------------------------------------------
+
+fetch_libmosquitto() {
+  download_tarball "${1}" "${2}" "libmosquitto" "mosquitto"
+}
+
+build_libmosquitto() {
+  run make -C "${1}/lib"
+}
+
+copy_libmosquitto() {
+  target_dir="${PWD}/mosquitto"
+
+  run mkdir -p "${target_dir}"
+
+  run cp "${1}/lib/libmosquitto.a" "${target_dir}"
+  run cp "${1}/lib/*.h" "${target_dir}"
+}
+
+bundle_libmosquitto() {
+  if [ -n "${NETDATA_DISABLE_LIBMOSQUITTO}" ] ; then
+    return 0
+  fi
+
+  progress "Prepare custom libmosquitto version"
+
+  MOSQUITTO_PACKAGE_VERSION="$(cat packaging/mosquitto.version)"
+
+  tmp=$(mktemp -d /tmp/netdata-mosquitto-XXXXXX)
```

Comment:
Why use such a specific template instead of `netdata-mosquitto-XXXXXX`. The manpage says that `/tmp` is assumed and there are situations where a user can put their temp storage somewhere else.

I can't see where the directory is cleaned up - why don't we delete it explicitly?

Reply:
Largely because that's how the Go plugin is doing it, and I just copied this code from there with minimal modification.

As far as cleanup, it isn't doing so for the moment simply because I forgot to add that.